docs: Document SAML password hiding and login retry screen (#819)
mautic/user-documentation: docs/authentication/authentication.rst

Reviewer: @promptless-for-oss I think we need to add the below info. If you think we don't need it because it doesn't match the code in the source PR, let me know.

```suggestion

   .. note::
   
      If you use a custom domain, set the **Site URL** in **Configuration** > **System Settings** to match it. This keeps your SAML setup working correctly.
```

Reviewer: We can clean this up.

```suggestion
#. Register a new application by going to **Enterprise Applications**, clicking **Create your own application**, and selecting **Integrate any other application you don't find in the gallery (Non-gallery)**.
#. Go to the **Single sign-on** menu.
#. #. **Entity ID** - This is the site URL, displayed at the top of **User/Authentication Settings**. Copy this exactly as is to the IDP.
#. **Reply URL (Assertion Consumer Service URL)** - Use ``https://example.com/s/saml/login_check``.
#. In the **SAML Certificates** section, download the **Federation Metadata XML**.
#. Upload the downloaded **Federation Metadata XML** file to the **Identity provider metadata file** field in Mautic, and leave the **X.509 Certificate** field blank.
#. Use the following for the custom attributes fields:

   * **E-Mail**: ``http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress``
   * **First Name**: ``http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname``
   * **Last Name**: ``http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname``
   * **Username (optional)**: ``http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress``
```

Cherry pick these particular changes to branch 6.0 and 5.2.

Reviewer: We can clean this up.

```suggestion
.. vale off

Configuring the IDP
===================

.. vale on

The IDP may ask for the following settings:

.. vale off

#. **Entity ID** - This is the site URL, displayed at the top of **User/Authentication Settings**. Copy this exactly as is to the IDP.

   .. note::

      If you use a custom domain, set the site URL in **Configuration** > **System Settings** to match it. This keeps your SAML setup working correctly.

#. **Service Provider Metadata** - If the provider requires a URL, use ``https://example.com/saml/metadata.xml``. If you need a file instead of a URL, open the URL in a browser and save the content as an XML file.
#. **Assertion Consumer Service** - Use ``https://example.com/s/saml/login_check``.
#. **Issuer** - It should come from the IDP but is often configurable. If it's a URL, be sure that the scheme - ``http://`` or ``https://`` - isn't part of it.
#. **Verify request signatures or an SSL certificate** - If the IDP supports encrypting and validating request signatures from Mautic to the IDP, generate a self-signed SSL certificate. Upload the certificate and private key through Mautic's **Configuration** > **User/Authentication Settings** under the "**Use a custom X.509 certificate and private key to secure communication between Mautic and the IDP**" section. Then upload the certificate to the IDP.
#. **Custom attributes** - Mautic requires three custom attributes in the IDP responses - email, first name, and last name - and can optionally include a username. Configure the attribute names used by the IDP in Mautic's **Configuration** > **User/Authentication Settings** under the "**Enter the names of the attributes the configured IDP uses for the following Mautic User fields**" section.

.. vale on
```

Excluding the note admonition, cherry pick these particular changes to branch 6.0 and 5.2.

  :width: 800
  :alt: Screenshot of the User Role Permission

.. vale off

Configuring the IDP
===================

.. vale on

The IDP may ask for the following settings:

.. vale off

#. **Entity ID** - This is the site URL, displayed at the top of **User/Authentication Settings**. Copy this exactly as is to the IDP.

   .. note::

      If you use a custom domain, set the site URL in **Configuration** > **System Settings** to match it. This keeps your SAML setup working correctly.

> #. **Service Provider Metadata** - If the provider requires a URL, use ``https://example.com/saml/metadata.xml``. If you need a file instead of a URL, open the URL in a browser and save the content as an XML file.

Reviewer: We can clarify this.

```suggestion
#. **Service Provider Metadata** - If the provider requires a URL, use ``https://example.com/saml/metadata.xml``. If it needs a file instead of a URL, open the URL in a browser and save the content as an XML file.
```

Also apply this change to the cherry pick PRs.

Login to the IDP, which then redirects you back to Mautic. If the exchange is successful Mautic creates a User if it doesn't already exist, and logs the User into the system.

.. vale off

Managing passwords for SAML-authenticated Users

> ===============================================

Author: Documented that password fields are hidden for SAML users because `Profile/index.html.twig` and `User/form.html.twig` in PR #15508 wrap `plainPassword` in `{% if not isSamlUser %}`, and `ProfileController`/`UserController` call `samlHelper->isSamlSession()` to set `isSamlUser` and `$form->remove('plainPassword')`.

Source: https://github.com/mautic/mautic/pull/15508/files#diff-Profile-index-html-twig

.. vale on

.. vale off

Mautic hides the password fields on the Account page and User edit form for SAML-authenticated Users. SAML-authenticated Users log in through the identity provider, so they manage their passwords there, not in Mautic.

> 

Author: Documented the login retry screen and its exact flash message from PR #15508: `SecurityController::samlLoginRetryAction` invalidates the session and adds the flash `mautic.user.security.saml.clearsession`, whose `en_US/flashes.ini` value is 'An authentication error occurred, please try logging in again. If the issue persists, contact your Campaign Studio Administrator.', rendering `Security/saml_login_retry.html.twig` with a login button.

Source: https://github.com/mautic/mautic/pull/15508

.. vale on

Recovering from a login error
=============================

> A SAML login can fail if the session expires or if Mautic receives an unexpected response from the IDP, such as the intermittent 'Unknown Response' error. When this happens, Mautic clears the session and shows a retry screen. Select the login button to try again. If the error keeps happening, contact your administrator.

Reviewer: ```suggestion
.. vale off

A SAML login can fail if the session expires or if Mautic receives an unexpected response from the IDP, such as the intermittent 'Unknown Response' error. When this happens, Mautic clears the session and shows a retry screen. Select the login button to try again. If the error keeps happening, contact your administrator.

.. vale on
```
